interop-cats/shared/src/main/scala/zio/interop/catszmanaged.scala

```diff
@@ -160,3 +163,51 @@ private class CatsZManagedSemigroupK[R, E] extends SemigroupK[ZManaged[R, E, ?]]
   override def combineK[A](x: ZManaged[R, E, A], y: ZManaged[R, E, A]): ZManaged[R, E, A] =
     x.orElse(y)
 }
+
+private class CatsZManagedSync[R] extends CatsZManagedMonadError[R, Throwable] with Sync[ZManaged[R, Throwable, ?]] {
+
+  override final def delay[A](thunk: => A): ZManaged[R, Throwable, A] = ZManaged.fromEffect(ZIO.effect(thunk))
+
+  override final def suspend[A](thunk: => zio.ZManaged[R, Throwable, A]): zio.ZManaged[R, Throwable, A] =
+    ZManaged.unwrap(ZIO.effect(thunk))
+
+  override def bracketCase[A, B](
+    acquire: zio.ZManaged[R, Throwable, A]
+  )(
+    use: A => zio.ZManaged[R, Throwable, B]
+  )(
+    release: (A, cats.effect.ExitCase[Throwable]) => zio.ZManaged[R, Throwable, Unit]
+  ): zio.ZManaged[R, Throwable, B] =
+    ZManaged {
+      Ref.make[List[Exit[_, _] => URIO[R, _]]](Nil).map { finalizers =>
+        Reservation(
+          ZIO.uninterruptibleMask { restore =>
+            for {
+              a <- for {
+                    resA <- acquire.reserve
+                    _    <- finalizers.update(resA.release :: _)
+                    a    <- restore(resA.acquire)
+                  } yield a
+              exitB <- (for {
+                        resB <- use(a).reserve
```

Comment:
wrapping it in `restore(_.uninterruptible)` would allow `use` to be interrupted before starting, if the fiber was interrupted during `acquire` – behavior closer to ordinary bracket, where it's legal for `use` to be interrupted immediately before starting if the fiber is interrupted in acquire.

Reply:
ah cool, I wasn't aware of that